spring/boot2-actuator-autoconfigure/src/main/java/com/linecorp/armeria/spring/actuate/ArmeriaSpringActuatorAutoConfiguration.java

```diff
@@ -217,7 +214,12 @@ ArmeriaServerConfigurator actuatorServerConfigurator(
                              // Bind a WebOperationService to "management.server.port"
                              sb.virtualHost(managementPort)
                                .service(route, new WebOperationService(operation, statusMapper));
-                         } else {
+                         }
+                         if (internalServicePort != null) {
```

Comment:
How about making a collection of ports and iterating them to create `WebOperationService`?
```java
List<Integer> ports = ...;

if (ports.isEmpty()) {
   // Bind to the default virtual host.
} else {
   for (int port in ports) {
       // Bind to a custom virtual host.
   }
}
```